pulp_2to3_migration/app/plugin/rpm/migrator.py

```diff
@@ -60,5 +87,95 @@ class RpmMigrator(Pulp2to3PluginMigrator):
         Migrate pre-migrated Pulp 2 RPM plugin content.
         """
         first_stage = ContentMigrationFirstStage(cls)
-        dm = DeclarativeContentMigration(first_stage=first_stage)
+        dm = RPMDeclarativeContentMigration(first_stage=first_stage)
         await dm.create()
+
+
+class RPMDeclarativeContentMigration(DeclarativeContentMigration):
+    """
+    A pipeline that migrates pre-migrated Pulp 2 rpm content into Pulp 3.
+    """
+
+    def pipeline_stages(self):
+        """
+        Build a list of stages.
+
+        This defines the "architecture" of the content migration to Pulp 3.
+
+        Returns:
+            list: List of :class:`~pulpcore.plugin.stages.Stage` instances
+
+        """
+        pipeline = [
+            self.first_stage,
+            QueryExistingArtifacts(),
+            ArtifactSaver(),
+            QueryExistingContents(),
+            ContentSaver(),
+            InterrelateContent(),
+            RelatePulp2to3Content(),
+        ]
+
+        return pipeline
+
+
+class InterrelateContent(Stage):
+    """
+    Stage for relating Content to other Content.
+    """
+
+    async def run(self):
+        """
+        Relate each item in the input queue to objects specified on the DeclarativeContent.
+        """
+        async for batch in self.batches():
+            modulemd_packages_batch = []
+            with transaction.atomic():
+                for dc in batch:
+                    if type(dc.content) == pulp3_models.Modulemd:
+                        thru = self.relate_package_to_module(dc)
+                        modulemd_packages_batch.extend(thru)
+
+                ModulemdPackages = pulp3_models.Modulemd.packages.through
+                ModulemdPackages.objects.bulk_create(objs=modulemd_packages_batch,
+                                                     ignore_conflicts=True,
+                                                     batch_size=1000)
+
+            for dc in batch:
+                await self.put(dc)
+
+    def relate_package_to_module(self, dc):
```

Comment:
Since it can't be generic and will always be a dc for modules, what about calling it `modulemd_dc`? for readability sake.
Then later, `modulemd_dc.content.artifacts` makes more sense to me :)

Reply:
i am ok with that. i have no issues here